pym/portage/emaint/modules/sync/sync.py

```diff
@@ -199,20 +167,58 @@ def _filter_auto(self, repos):
 	def _sync(self, selected_repos, return_messages,
 		emaint_opts=None):
 
+		# No repos defined or no auto-sync repos when the caller is auto_sync().
+		if not selected_repos:
+			if return_messages:
+				return (True, ["Nothing to sync... returning"])
+			return (True, None)
+
 		if emaint_opts is not None:
 			for k, v in emaint_opts.items():
 				if v is not None:
 					k = "--" + k.replace("_", "-")
 					self.emerge_config.opts[k] = v
 
-		selected_repos = [repo for repo in selected_repos if repo.sync_type is not None]
 		msgs = []
-		if not selected_repos:
-			msgs.append("Emaint sync, nothing to sync... returning")
+		if self.emerge_config.args:
+			emerge_repos = []
+			unknown_repo_names = []
+			skipped_repo_names = []
+			auto_sync_repos = {repo.name: repo for repo in selected_repos}
+			for repo_name in self.emerge_config.args:
+				#print("_sync(): repo_name =", repo_name)
+				try:
+					repo = self.emerge_config.target_config.settings.repositories[repo_name]
+				except KeyError:
+					unknown_repo_names.append(repo_name)
+				else:
+					if repo_name not in auto_sync_repos:
+						skipped_repo_names.append(repo_name)
+					else:
+						emerge_repos.append(auto_sync_repos[repo_name])
+			if unknown_repo_names:
+				msgs.append(warn(" * ") + "Unknown repo(s): %s\n" %
+					" ".join(unknown_repo_names));
+			if skipped_repo_names:
+				msgs.append(warn(" * ") + "auto-sync is disabled for repo(s): %s" %
+					" ".join(skipped_repo_names) + ", skipping...\n")
+			selected_repos = emerge_repos
+
+		valid_repos = []
+		missing_sync_type = []
+		for repo in selected_repos:
+			if repo.sync_type is not None:
+				valid_repos.append(repo)
+			else:
+				missing_sync_type.append(repo.name)
+		if missing_sync_type:
+			msgs.append(warn(" * ") + "Missing sync-type for repo(s): %s" %
+				" ".join(missing_sync_type) + ", skipping...\n")
+		if not valid_repos:
```

Comment:
It only fails if valid_repos is empty? I want it to fail if there are __any__ unknown repos or missing sync types for repos that are supposed to be synced.